src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/FaultExtensions.cs

```diff
@@ -171,7 +174,58 @@ internal static class FaultExtensions
                 options &= ~ForkOptions.CancelOnUnload;
             }
 
-            threadingService.Fork(asyncAction, factory: null, configuredProject: configuredProject, watsonReportSettings: s_defaultReportSettings, faultSeverity: faultSeverity, options: options);
+            threadingService.Fork(asyncAction, factory: null, configuredProject: configuredProject, watsonReportSettings: s_defaultReportSettings, faultSeverity: severity, options: options);
+        }
+
+        /// <summary>
+        ///     Attaches error handling to a block so that if it throws an unhandled exception,
+        ///     the error will be reported to the user.
+        /// </summary>
+        /// <param name="faultHandlerService">
+        ///     The <see cref="IProjectFaultHostHandler"/> that should handle the fault.
+        /// </param>
+        /// <param name="block">
+        ///     The block to attach error handling to.
+        /// </param>
+        /// <param name="project">
+        ///     The project related to the failure, if applicable. Can be <see langword="null"/>.
+        /// </param>
+        /// <param name="severity">
+        ///     The severity of the failure.
+        /// </param>
+        public static Task RegisterFaultHandler(
+            this IProjectFaultHandlerService faultHandlerService,
+            IDataflowBlock block,
+            UnconfiguredProject? project,
+            ProjectFaultSeverity severity = ProjectFaultSeverity.Recoverable)
+        {
+            Requires.NotNull(faultHandlerService, nameof(faultHandlerService));
+            Requires.NotNull(block, nameof(block));
+
+            return block.Completion.ContinueWith(_ => 
```

Comment:
I believe this continuation will only stay in the memory when the block is chained. Otherwise, the entire thing will be GCed. Just want to conform that.

Reply:
How did you determine that? This just hooks up to the underlying TaskCompletionSource: http://index/?leftProject=Microsoft.VisualStudio.ProjectSystem&leftSymbol=fy8f3jrwsdxu&file=Core%5CDataflow%5CDataflowBlockSlimBase.cs&line=86.